extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/querykit/groupby/GroupByQueryKit.java

```diff
@@ -132,12 +133,12 @@ public QueryDefinition makeQueryDefinition(
       shuffleSpecFactoryPreAggregation = intermediateClusterBy.isEmpty()
                                          ? ShuffleSpecFactories.singlePartition()
                                          : ShuffleSpecFactories.globalSortWithMaxPartitionCount(maxWorkerCount);
-      shuffleSpecFactoryPostAggregation = doLimitOrOffset
+      shuffleSpecFactoryPostAggregation = doLimitOrOffset && !needsFinalShuffling
                                           ? ShuffleSpecFactories.singlePartition()
                                           : resultShuffleSpecFactory;
       partitionBoost = true;
     } else {
-      shuffleSpecFactoryPreAggregation = doLimitOrOffset
+      shuffleSpecFactoryPreAggregation = doLimitOrOffset && !needsFinalShuffling
```

Comment:
This change looks incorrect - Irrespective of whether we require final shuffling or not, to compute the limit properly, we need to generate a single partition.

Reply:
Removed this parameter